core/src/main/java/cucumber/runtime/Runtime.java

```diff
@@ -55,7 +57,7 @@ private Runtime(final Plugins plugins,
         exitStatus.setEventPublisher(bus);
     }
 
-    public void run() {
+    public void run() throws InterruptedException {
```

Comment:
I inlined the runner. Keeps it clear what the exact process is.

Reply:
Fair enough, I was just trying to reduce the cyclomatic complexity of the class.  Also by `invokeAll` it allowed me to easily wrap the `InterruptedException` into a `CucumberException`

Reply:
Good point. The threaded stuff is an implementation detail.